platform/src/main/java/org/hillview/maps/ToCatMap.java

```diff
@@ -0,0 +1,67 @@
+package org.hillview.maps;
+
+import org.hillview.dataset.api.IMap;
+import org.hillview.table.CategoryArrayColumn;
+import org.hillview.table.CategoryListColumn;
+import org.hillview.table.ColumnDescription;
+import org.hillview.table.Table;
+import org.hillview.table.api.ContentsKind;
+import org.hillview.table.api.IColumn;
+import org.hillview.table.api.IRowIterator;
+import org.hillview.table.api.ITable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This map receives a column name of the input table, and returns a table with the same column, with additionally
+ * that specified column converted to a categorical column.
+ */
+public class ToCatMap implements IMap<ITable, ITable> {
+    private final String inputColName;
+    private final String catColName;
+
+    /**
+     * @param inputColName The name of the column that has to be converted to a categorical column.
+     */
+    public ToCatMap(String inputColName) {
+        this.inputColName = inputColName;
+        this.catColName = inputColName + " (Cat.)";
```

Comment:
The output column name should be generated by the UI and be a parameter.
Maybe you already have a column with this name. You have to make sure that the name is unique.